packages/fluentui/react/src/themes/teams/components/Checkbox/checkboxStyles.ts

```diff
@@ -80,53 +116,61 @@ const checkboxStyles: ComponentSlotStylesPrepared<CheckboxStylesProps, CheckboxV
     ...(p.disabled &&
       p.checked && {
         color: v.disabledCheckedIndicatorColor,
-        background: v.disabledBackgroundChecked,
-        borderColor: 'transparent'
+        borderColor: v.disabledBackgroundChecked,
+        backgroundImage: checkboxIndicatorUrl(v.disabledCheckedIndicatorColor, v.disabledBackgroundChecked)
       })
   }),
 
   toggle: ({ props: p, variables: v }): ICSSInJSStyle => ({
     '-ms-grid-row-align': 'center',
     gridColumn: p.labelPosition === 'start' ? 3 : 1,
     boxShadow: 'unset',
+    boxSizing: 'border-box',
 
     background: v.background,
     borderColor: v.borderColor,
     borderStyle: v.borderStyle,
     borderRadius: v.toggleBorderRadius,
     borderWidth: v.borderWidth,
-    color: v.borderColor,
     margin: v.toggleMargin,
-    padding: v.togglePadding,
-    transition: 'padding .3s ease',
     userSelect: 'none',
     width: v.toggleWidth,
     height: v.toggleHeight,
 
-    [`& svg`]: {
-      width: v.toggleIndicatorSize,
-      height: v.toggleIndicatorSize
+    ':before': {
+      ...commonToggleBeforeStyles(v),
+      borderColor: p.disabled ? v.disabledToggleIndicatorColor : v.borderColor,
+      borderStyle: v.borderStyle,
+      borderWidth: v.borderWidth,
+      margin: v.togglePadding
     },
 
-    ...(p.checked && {
-      background: v.checkedBackground,
-      borderColor: v.checkedBorderColor,
-      color: v.checkedIndicatorColor,
-      padding: v.toggleCheckedPadding
-    }),
+    ...(p.checked &&
+      !p.disabled && {
+        borderColor: v.checkedBorderColor,
+        backgroundColor: v.checkedBackground,
+        ':before': {
+          ...commonToggleBeforeStyles(v),
+          margin: v.toggleCheckedPadding,
+          background: v.checkedIndicatorColor
+        }
+      }),
 
     ...(p.disabled && {
-      color: v.disabledToggleIndicatorColor,
-      background: v.disabledBackground,
-      borderColor: v.disabledBorderColor
-    }),
-
-    ...(p.disabled &&
-      p.checked && {
-        color: v.disabledCheckedIndicatorColor,
+      ...(!p.checked && {
```

Comment:
I don't understand this... Why `!p.checked` is required there? Should it be just:

```
...(p.disabled && {
  background: v.disabledBackground,
  borderColor: v.disabledBorderColor
  ...(p.checked && {
    background: v.disabledBackgroundChecked,
    borderColor: 'transparent',
  }),
}
```